zerver/lib/upload.py

```diff
@@ -353,6 +353,9 @@ def upload_avatar_image(self, user_file, user_profile, email):
         resized_medium = resize_avatar(image_data, MEDIUM_AVATAR_SIZE)
         write_local_file('avatars', email_hash+'-medium.png', resized_medium)
 
+        user_profile.avatar_version += 1
```

Comment:
For the use case of bots, the `user_profile` here refers to the human owner of the bot, not the bot user.  We actually want to bump the avatar version for the bot user.  For this reason we probably want to change the parameters here to be `owner_profile` and  `user_profile`, and then we'll need to fix the calling code and appropriate tests.

There's not a a lot of code to write for this particular feature, but it's really important to read the code here to understand the big picture, and it may also be helpful to learn how our bot setup works.  In your local environment you can try creating a few bots and fixing their avatars to see how bots work.